core/indexer/processTransactions.go

```diff
@@ -196,19 +196,41 @@ func (tdp *txDatabaseProcessor) addScResultInfoInTx(scHash string, scr *smartCon
 	tx.SmartContractResults = append(tx.SmartContractResults, dbScResult)
 
 	if isSCRForSenderWithGasUsed(dbScResult, tx) {
-		gasUsed := tx.GasLimit - scr.GasLimit
-		tx.GasUsed = gasUsed
+		tx.GasUsed = computeTxGasUsedField(dbScResult, tx)
 	}
 
 	return tx
 }
 
+func computeTxGasUsedField(dbScResult ScResult, tx *Transaction) uint64 {
+	fee := big.NewInt(0).SetUint64(tx.GasPrice)
+	fee.Mul(fee, big.NewInt(0).SetUint64(tx.GasLimit))
+
+	refundValue, ok := big.NewInt(0).SetString(dbScResult.Value, 10)
+	if !ok {
+		return fee.Uint64()
```

Comment:
is this ok?

Reply:
is there a case that the value is invalid and the fee must be this? or is it just not to return error?

Reply:
No. This case will never happen because `dbScResult.Value` is set from a `big.Int` so this `SetString` will never fail.

Reply:
ok. At least set a log warn. Who knows?

Reply:
done.